### PR TITLE
Upgrade polkadot sdk v1.13.0

### LINKED ISSUE #205: 🎁 [Feature Request]: upgrade polkadot-sdk to `v1.13.0`
### templates

- [X] Generic Runtime Template
- [X] Evm Template

### What is the feature you would like to see?

1. choose polkadot-sdk version to upgrade before/until higher priority issues are done. `v1.12.0` was released 5 days ago as of writing this
2. upgrade all forks required for external dependencies 
3. document and iteratively automate all processes

(1) requires beforehand agreement as a group to choose the stable version. The work required for (2) will determine timing and prioritization of (1).

It may not be ideal to be the first to upgrade all the dependencies i.e. frontier as well as any additional external dep forks required to finish #189 i.e. ORML, Acala, Moonbeam, HydraDX

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by ozgunozerk**
Fixes #205 .

Thanks to @4meta5 for updating `Moonbeam` and `ORML` forks,
Thanks to @KitHat for helping me tackle with the compilation errors related to `evm/node` and `fuzzers`.

This was probably the hardest upgrade we have done so far, great team work! 🎉

### CODE REVIEW

File: evm-template/runtime/src/weights/pallet_balances.rs [hunk 1]
```diff
@@ -68,6 +68,28 @@ impl<T: frame_system::Config> pallet_balances::WeightInfo for WeightInfo<T> {
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
+	//	TODO: below is random
```
Comment (ozgunozerk) [inline evm-template/runtime/src/weights/pallet_balances.rs:71]: @KitHat I had to add these random entries to make the compiler happy. Do you think we should address these random values immediately, or they can wait till the next time we update our benchmarks?

Comment (KitHat) [reply evm-template/runtime/src/weights/pallet_balances.rs:71]: We can leave them for now and address them during the release process

File: evm-template/node/src/rpc/eth.rs [hunk 2]
```diff
@@ -40,13 +40,13 @@ pub struct EthDeps<B: BlockT, C, P, A: ChainApi, CT, CIDP> {
     /// Whether to enable dev signer
     pub enable_dev_signer: bool,
     /// Network service
-    pub network: Arc<NetworkService<B, B::Hash>>,
+    pub network: Arc<dyn NetworkService>,
```
Comment (KitHat) [inline evm-template/node/src/rpc/eth.rs:43]: Is there any reason to mix generics and trait objects?

Comment (ozgunozerk) [reply evm-template/node/src/rpc/eth.rs:43]: Upstream has refactored Network service, hence these changes were also introduced in upstream. I simply followed them tbh

**Review by KitHat — APPROVED** [review]

**Pull request merged**